### ISSUE TITLE
Model 'array' attribute type doesn't work

### ISSUE THREAD

**Issue by pablovilas** [balderdashy/sails-mysql#168]
'array' attribute type is not working for sails-mysql, throws this error:  "Error: ".

**Comment by devinivy** [comment]
If you provide a code example, maybe we could help make the adapter provide a more useful error in that scenario.

**Comment by pursual** [comment]
I dont get an error, but trying to query on an array field produces an "IN" statement that does not work. It should maybe produce a "find in set" statement?

**Comment by devinivy** [comment]
Supporting FIND_IN_SET for array types would be a nice improvement.  Would be a breaking change, as we currently store array types as stringified JSON.  Might be some hard edge-cases.

**Comment by pursual** [comment]
I don't care how they are saved as long as they are queryable. But in the end it looks like with SQL the better route is associations.

**Comment by devinivy** [comment]
This is why arrays are prioritized as queryable.  There's always the possibility of using associations for more complex use-cases.  This will be especially awesome once deep population is supported by waterline.

**Comment by pursual** [comment]
Arrays are not queryable in SQL though, correct? Unless you implement a raw "find in set".

**Comment by sailsbot** [comment]
Thanks for posting, @pablovilas.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
